### PR TITLE
Setting up a home energy monitoring system

### PR DESCRIPTION
**Pull request by eloots**

### CODE REVIEW

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc [hunk 1]
```diff
@@ -0,0 +1,711 @@
...
+  topics = [
+    "servicelocation/5aaf6e89-89cb-4e33-bf34-05abc62f5563/realtime"
+  ]
+  # The "host" tag is irrelevant in this use case. Drop it
+  tagexclude = ["host"]
+  data_format = "json_v2"
+  [[inputs.mqtt_consumer.json_v2]]
+    [[inputs.mqtt_consumer.json_v2.field]]
+      path = "channelPowers.#(formula==$5500048161/0$).power"
+      rename = "zolder-verlichting"
```
Comment (LeonorLunatech) [inline posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:319]: can you use the same english names you used above? `lighting` , `bicycle garage`, `sockets-network-switch`?

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc [hunk 2]
```diff
@@ -0,0 +1,711 @@
...
+v1.0, 2023-06-09
+:title: Setting up a home energy monitoring system
+:imagesdir: ../media/2023-06-09-setting-up-a-home-energy-monitoring-system
+:lang: en
+:tags: [raspberry pi, emqx, influxdb, smappee, telegraf, grafana, energy, monitoring]
+
+[id=introduction]
+== Introduction
+
+We had been using a https://www.smappee.com/infinity[_Smappee_] system that came with our solar panel installation at home for about two years. It was installed back in July 2020 to monitor the electricity generation and consumption in our house when I noticed something odd. I used the neat _Smappee_ iOS app to check the five-minute averages to monthly averages of the energy generated by the solar panels on our roof. The system was tracking granular numbers for longer than two weeks until it abruptly changed it to two weeks.
```
[the remark this answers is not in the document]
Comment (eloots) [reply posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:12]: I see your point... I will rephrase it to make it more clear.
In summary:
- Before: track 5 minute averages for longer than 2 weeks
- After: after 2 weeks, only provide 1h averages

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc (hunk 1 above)
Comment (eloots) [reply posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:319]: Will do.

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc [hunk 3]
```diff
@@ -0,0 +1,711 @@
...
+
+For example, a _Smappee_, when configured to send data to an _MQTT Broker_ sends data at a one-second interval. One may want to aggregate this data at minutes, days, weeks, or months intervals. Each aggregation level may have specific data retention periods. For example, we may keep the high frequency 1 minute aggregates for a few months. On the other side of the spectrum, we may keep indefinitely the monthly averages (and obviously, as long as storage capacity permits).
+
+As I wanted to be independent of external factors (such as the _Smappee_ Cloud) as to not be exposed to the whims of external parties, I chose the _MQTT_ route for building my _HEMS_.
+
+
+Next to the already cited main objective, there were other objectives I wanted to achieve:
+
+* If possible, implement the system without any programming.
+* Use _Open Source Software_ (OSS) only.
```
Comment (mcsuka) [inline posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:49]: OSS in the IT system management and monitoring world usually means `Operational Support System`. Your HEMS can be considered as an Operational Support System. I think using `FOSS (Free Open Source Software)` here would make this distinction clear.

Comment (eloots) [reply posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc]: Changed `OSS` to `FOSS` everywhere in the doc.

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc [hunk 4]
```diff
@@ -0,0 +1,711 @@
...
+
+After an initial spike (with a peak of 2kW) for about 20 minutes, the system evolves to a steady state where it consumes about 0.4kW for about 4 hours, after which changes to a kind of on-off mode.
+
+As for the second question, we can have a look at the following graph with data from May 15^th^:
+
+image::power-profile-electrical-bike-charging.png[Power consumption profile for charging electrical bicycle]
+
+We see that the power at which the battery is charged rises from an initial value of ±160W to 180W over a period of almost 2 hours. I think this is because it wasn't very warm outside, the battery's temperature rises during charging and the charging current is probably a function of that temperature, but that's an unconfirmed and personal theory. After reaching the peak, the charging slows down and stops after another one and a half hour.
+
+In conclusion, we can learn a lot about the different electrical consumers in the house and even learn a few special things about certain devices like the air conditioning.
```
Comment (gianick) [inline posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:618]:
```suggestion
In conclusion, we can learn a lot about the different electrical consumers in the house and even learn a few special things about certain devices like the air-conditioning.
```

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc [hunk 5]
```diff
@@ -0,0 +1,711 @@
...
+* Apparent and Real power 
+* cos(&#966;)
+
+We can have a look at the data for the garage when the hybrid car is charged:
+
+image::charging-hybrid-car-power.png[Power consumption during charging of hybrid car]
+
+The apparent power and real power are 2365 VA and 2357 W respectively. This corresponds to a cos(&#966;) = 0.997 which is pretty much close to ideal. This is especially important as the power consumed during the charging is about 2.4kW. 
+
+On the other hand, the charger of our electrical bicycle charger's cos(&#966;) is only 0.69. So, when charging, the different powers are P~real~ = 180W, P~a~ = 260VA, and P~reactive~ = 187W. Even though it's "only 187W", when thousand or millions of devices with a low Power factor are online, the impact _is_ significant.
```
Comment (gianick) [inline posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:633]: Just want to point that in this sentence you are using no space between the value and the unit `W`, `VA`.

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc [hunk 6]
```diff
@@ -0,0 +1,711 @@
...
+In conclusion, we can learn a lot about the different electrical consumers in the house and even learn a few special things about certain devices like the air conditioning.
+
+=== The Power Quality dashboard
+
+This dashboard contains two graphs that display the following data for all channels on the _Smappee Infinity_:
+
+* Apparent and Real power 
+* cos(&#966;)
+
+We can have a look at the data for the garage when the hybrid car is charged:
```
Comment (gianick) [inline posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:627]:
```suggestion
We can have a look at the data for the garage when the hybrid car is being charged:
```
Are you referring to the period during which the car is being charged or after it's been charged?

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc [hunk 7]
```diff
@@ -0,0 +1,711 @@
...
+
+<<<
+[id=cos-phi]
+=== Electrical Power Factor (also known as cos(&#966;))
+
+The Power Factor is a measure of an electrical system's energy utilisation efficiency. The power consumption can be decomposed in three parts:
+
+* P~a~: Apparent power, which is the product of the measured voltage and current.
+
+* P~real~: Effective- or Real Power is the part that actually produces work in the broad sense of the word, it's not the internal efficiency of the device itself. For example, an incandescent lightbulb consuming 100W converts 98W to heat and only 2W to light. So, in this case, the Productive Power is 100W (this is not to be confused with the efficiency of that bulb which is 2%).
```
Comment (gianick) [inline posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:643]:
```suggestion
* P~real~: Effective or Real Power is the part that actually produces work in the broad sense of the word, it's not the internal efficiency of the device itself. For example, an incandescent lightbulb consuming 100W converts 98W to heat and only 2W to light. So, in this case, the Productive Power is 100W (this is not to be confused with the efficiency of that bulb which is 2%).
```
Do we need the `-` there?

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc [hunk 8]
```diff
@@ -0,0 +1,711 @@
...
+
+* P^2^~a~ = P^2^~real~ + P^2^~reactive~
+
+* P~real~ = P~a~ . cos(&#966;)
+
+* P~reactive~ = P~a~ . sin(&#966;)
+
+(If you're interested about the theory around this topic, read link:../media/2023-05-05-setting-up-a-home-energy-monitoring-system-power-factor.pdf[this article] about it.)
+
+In electrical systems, cos(&#966;) is a value between 0 and 1. When the reactive power is 0, cos(&#966;) is equal to 1 and P~a~ and P~real~ have equal values.
```
Comment (gianick) [inline posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:657]:
```suggestion
In electrical systems, cos(&#966;) is a value between 0 and 1. When the Reactive power is 0, cos(&#966;) is equal to 1 and P~a~ and P~real~ have equal values.
```
Should this be `reactive` or `Reactive`?

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc [hunk 9]
```diff
@@ -0,0 +1,711 @@
...
+
+On the other extreme, P~reactive~ is equal to P~a~, and P~real~ and cos(&#966;) are both 0.
+
+Both the real- and the reactive power components transfer energy between energy producer and consumer. The real power component corresponds to a unidirectional transfer of energy from producer to consumer. The reactive power component corresponds to energy being bounced back and forth between producer and consumer.
+
+If we look at the bigger picture, both are transferred through the grid via high voltage transmission lines, transformers, and local power distribution systems. During this transfer, losses occur amounting to 6% to 8% of the total energy produced.
```
Comment (gianick) [inline posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:663]:
```suggestion
If we look at the bigger picture, both are transferred through the grid via high-voltage transmission lines, transformers, and local power distribution systems. During this transfer, losses occur amounting to 6% to 8% of the total energy produced.
```

File: media/2023-06-09-setting-up-a-home-energy-monitoring-system/08-16days.PNG
[the remark this answers is not in the document]
Comment (eloots) [reply media/2023-06-09-setting-up-a-home-energy-monitoring-system/08-16days.PNG:1]: Yes, that may be the case and we'll find out as soon as it gets rendered the first time in acceptance.
Changing the image to having 5 x 3 + 1 from 4 x 4 may be the way to go.

File: media/2023-06-09-setting-up-a-home-energy-monitoring-system/2023-06-09-setting-up-a-home-energy-monitoring-system-power-factor.tex [hunk 10]
```diff
@@ -0,0 +1,328 @@
...
+\begin{itemize}
+	\item $\theta$ = 0 for a resistive load. Examples of this an electrical heater system or an incandescent lightbulb.
+	\item $\theta$ $>$ 0 for inductive loads. The current lags the voltage. This is the case for electrical motors or coils.
+	\item  $\theta$ $<$ 0 for capacitive loads. The current precedes the voltage. This is the case for capacitors and battery chargers.
+\end{itemize}\
+
+In the most general case, a load acts as a combination of a resistive 
+and a reactive part, where the latter is either capacitive or inductive.\
+
+The following graph show a line voltage and a current plotted over time. In
```
[the remark this answers is not in the document]
Comment (eloots) [reply media/2023-06-09-setting-up-a-home-energy-monitoring-system/2023-06-09-setting-up-a-home-energy-monitoring-system-power-factor.tex:50]: Still amazed that I read over these kind of things even after proofreading them multiple times...

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc (hunk 5 above)
Comment (eloots) [reply posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:633]: I'll do a scan of the full document once I looked at all your suggestions and comments. I need to follow a single conventions (either no space of exactly one space between value and unit)

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc (hunk 7 above)
Comment (eloots) [reply posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:643]: This is actually a literal transplantation of a language construct in Dutch:
In Dutch, instead of writing: _Effective Power or Real Power is_
we write _Effective**-** or Real Power is_ where the hyphen is a placeholder for _Power_ .
I just tried to find if English has something similar, but apparent it hasn't. I'll commit your suggestion.

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc (hunk 8 above)
Comment (eloots) [reply posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:657]: For consistency, going of upper caps version

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc [hunk 11]
```diff
@@ -0,0 +1,711 @@
...
+
+To illustrate the issue, let’s look at a couple of sample plots. The first one is an example of 5 minute average production and consumption data for one day (August 23, 2022):
+
+image::23-08.PNG[5 minute averages over one day on August 23rd]
+
+The next one shows the same graphs over a period of 16 days (from August 8th till August 23rd):
+
+image::08-16days.PNG[Production and consumption data over 16 consecutive days]
+
+The graph shows the granularity changes from 5 minutes to an hour on August 8th.
```
[the remark this answers is not in the document]
Comment (eloots) [reply posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:22]: Actually, I'm going to change illustration completely by eliding most on the graphs on intermediate dates as these don't really add a lot of information.

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc [hunk 12]
```diff
@@ -0,0 +1,711 @@
...
+
+The obvious question I asked myself was how to keep granular data longer, and I saw two options. First, I could ask the company behind the system to reinstate the longer data retention period. Second, I could investigate if I could build an energy monitoring system that would allow me to configure the data retention policy as I wished it to be. I chose option two, as it seemed to give me an opportunity to learn technologies I had never used before.
```
[the remark this answers is not in the document]
Comment (eloots) [reply posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:24]: Hesitating about this one...

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc [hunk 13]
```diff
@@ -0,0 +1,711 @@
...
+When using _MQTT_, the _Smappee_ monitoring system configuration has to be tweaked to point to a so-called _MQTT Broker_. Once that’s done, the _Smappee_ device will send measurement samples at a once per second interval to the _MQTT Broker_.
+
+The first access method polls data from the _Smappee_ Cloud, while the second pushes data to a server.
+
+With all this in mind, let’s consider the project goals I set for my Home Energy Monitoring System (_HEMS_).
+
+[id=project-objectives]
+== Project objectives
+
+The primary aim of the _HEMS_ project was to collect _Smappee_ data at specific aggregation levels and retention periods. Also, we want dashboards that display selected aggregates over a chosen time period.
```
[the remark this answers is not in the document]
Comment (eloots) [reply posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:39]: Correct. I will change the "We" form to the "I" variant.

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc [hunk 14]
```diff
@@ -0,0 +1,711 @@
...
+```toml
+[[processors.regex]]
+  namepass = ["smappee-data-2"]
+  [[processors.regex.tags]]
+    key = "topic"
+    pattern = ".*/(.*)/.*"
+    replacement = "smappee/${1}"
+```
+
+We can be observe:
```
[the remark this answers is not in the document]
Comment (eloots) [reply posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:358]: Of course :-)
Will change it.

[the remark this answers is not in the document]
Comment (eloots) [reply posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:358]: And a good catch!

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc [hunk 15]
```diff
@@ -0,0 +1,711 @@
...
+
+I spent little time securing the set-up as I think InfluxDB version 2 has a lot more to offer.
+
+Actually, when I started this project, I wasn’t aware of the fact that there is a version 2 of InfluxDB. I found out by the time the project was already long underway. I did a small trial by uninstalling version 1 followed by an installation of version 2. What I found impressive is that when I started the version 2 instance for the first time, it told me it had detected version 1 databases and if I wanted them converted to version 2. I accepted the offer and it worked flawlessly. What I liked even more is that when I reverted the installation to version 1, my original data was still there and ready to continue where I left off. A really refreshing experience!
+
+=== Provisioning the _HEMS_ system with _cloud-init_
+
+_cloud-init_ is a method for cross-platform instance initialisation. We can utilise it even on bare-metal installations like on a Raspberry Pi. It performs user creation, can execute custom scripts, installs packages, creates files, partitions disk, creates file systems, etc.
```
[the remark this answers is not in the document]
Comment (eloots) [reply posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:448]: I prefer to use British-English. Probably needs another scan to check occurrences of American-English.

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc [hunk 16]
```diff
@@ -0,0 +1,711 @@
...
+
+A _cloud-init_ deployment is driven by a cloud-config file in YAML format. You can find the configuration for this project https://github.com/eloots/home-energy-monitoring-system-setup/blob/main/cloud-init/smappee-2.yml[here]. It's part of the _HEMS_ GitHub repository that also contains the Telegraf configuration https://github.com/eloots/home-energy-monitoring-system-setup/blob/main/telegrafConfiguration/etc/telegraf/telegraf.d/smappee-2.conf[smappee-2.conf] for the _Smappee Genius_ and https://github.com/eloots/home-energy-monitoring-system-setup/blob/main/telegrafConfiguration/etc/telegraf/telegraf.d/smappee.conf[smappee.conf] for the _Smappee Solar_.
+
+Noteworthy mentioning is that _cloud-init_ supports instance data with [jinja] template rendering. Instead of directly applying configuration settings in the _cloud-config_ file, metadata comprising key/value pairs can be passed to _cloud-init_ in a file and these can be de-referenced in the _cloud-config_ file.
```
[the remark this answers is not in the document]
Comment (eloots) [reply posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:458]: Good find. Will add it.

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc [hunk 17]
```diff
@@ -0,0 +1,711 @@
...
+image::Grafana-aggregate-consumption-production.png[Smappee connection diagram]
+
+The six status panels at the top display the following information:
+
+* Total solar energy generated over the selected time period in kWh.
+* Average Solar power generated in kW over the last minute.
+* Total energy consumption over the selected time period in kWh.
+* Average total energy consumption in kW over the last minute.
+* Mains voltage minimum and maximum value over the select time period in Volts.
+* Mains frequency in Herz (this value is measured by the _Smappee Genius_).
```
[the remark this answers is not in the document]
Comment (eloots) [reply posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:496]: Apparently and surprisingly (after looking it up in the Collins dictionary), it's _hertz_ (small caps).

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc [hunk 18]
```diff
@@ -0,0 +1,711 @@
...
+Even though the installed Solar panel capacity is about 4 kW, the production capacity is capped to 2500W by the lower rated power of the inverter attached to the panels.
+
+The Grafana dashboard allows us to quickly answer questions like:
+
+* What is the total amount of energy produced by the solars panels?
+* What is the peak power produced by the solar panels?
+* What is the total amount of energy produced by the solar panels during ramp-up in the morning till 13:00?
+* What is the total amount of energy produced by the solar panels for the remainder of the day?
+
+The top-left status panel tell us that the answer to the first question is 18.79 kWh.
```
[the remark this answers is not in the document]
Comment (eloots) [reply posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:515]: It's specifically the [single] top left panel

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc [hunk 19]
```diff
@@ -0,0 +1,711 @@
...
+
+image::Grafana-finding-ratios-3-4-5.png[Finding production ratios - 3-4-5]
+
+The amount of energy produced in each section of the production curve is 4.21 kWh, 12.75 kWh, and 2.97 kWh. When we express this as relative to total production number, we get the following ratios:
+
+* Ramp-up: 21.1%
+* Steady state: 64.0%
+* Ramp-down: 14.9%
+
+The beautiful thing about this is that this exploration and calculation takes just a minute to complete.
```
[the remark this answers is not in the document]
Comment (eloots) [reply posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:586]: Thanks, opting for your second suggestion.

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc [hunk 20]
```diff
@@ -0,0 +1,711 @@
...
+
+image::power-profile-airco.png[Power consumption profile for airco]
+
+After an initial spike (with a peak of 2kW) for about 20 minutes, the system evolves to a steady state where it consumes about 0.4kW for about 4 hours, after which changes to a kind of on-off mode.
+
+As for the second question, we can have a look at the following graph with data from May 15^th^:
+
+image::power-profile-electrical-bike-charging.png[Power consumption profile for charging electrical bicycle]
+
+We see that the power at which the battery is charged rises from an initial value of ±160W to 180W over a period of almost 2 hours. I think this is because it wasn't very warm outside, the battery's temperature rises during charging and the charging current is probably a function of that temperature, but that's an unconfirmed and personal theory. After reaching the peak, the charging slows down and stops after another one and a half hour.
```
[the remark this answers is not in the document]
Comment (eloots) [reply posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:616]: Will repeat `±`.

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc (hunk 12 above)
[the remark this answers is not in the document]
Comment (eloots) [reply posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc]: Changed to what you suggested, except that I use a regular item list instead of a numbered one.

**Review by gianick — APPROVED** [review]

**Review by thinkmorestupidless — COMMENTED** [review]
This is a BIG post... there's a ton of detail here and it's really impressive...

Apart from the small copy suggestions i'd make two suggestions to the structure... 

1. Perhaps provide a 'table of contents' type section at the beginning, linking to each major section, perhaps... the point being if the sections are laid out together this might help paint the picture of how the post will progress (and also make it easier to return to if it's not consumed in a single sitting).
2. The 'Grafana' section at the end... could this be something that's split into a separate post? A companion post that just focusses on the dashboard - letting the reader focus on the detailed implementation explanation/discussion/review in this post and then see the visualisations in the follow-up... i don't have strong opinion enough to force the issue, but if you're thinking about splitting it this is where it might make sense...

Excellent stuff, though, love it 👍

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc [hunk 21]
```diff
@@ -0,0 +1,718 @@
...
+
+* A Raspberry Pi 4 Model B/8GB with a 250GB SSD (SATA disk connected to one of the Pi's USB-3 ports via a USB-SATA converter).
+
+* The software components &#8212; EMQX MQTT Broker, Telegraf, InfluxDB server, and Grafana Server with data flowing from left to right.
+
+* An _MQTT_ client &#8212; mainly used during debugging the _MQTT_/Telegraf configuration. The EMQX project has an _MQTT_ client with a Graphical User Interface named https://github.com/emqx/MQTTX[MQTTX], but due to it being pretty slow, I switched to https://github.com/eclipse/mosquitto[Mosquitto] CLI.
+
+* I added the Raspberry Pi OS Metrics Telegraf configuration and Grafana Dashboard to monitor the Raspberry Pi.
+
+With this out of the way, let's look at configuration and system provisioning in the next chapters.
```
[the remark this answers is not in the document]
Comment (eloots) [reply posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:163]: Weird; I thought I had that changed already (as suggested by @gianick)

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc [hunk 22]
```diff
@@ -0,0 +1,718 @@
...
+v1.0, 2023-06-09
+:title: Setting up a home energy monitoring system
+:imagesdir: ../media/2023-06-09-setting-up-a-home-energy-monitoring-system
+:lang: en
+:tags: [raspberry pi, emqx, influxdb, smappee, telegraf, grafana, energy, monitoring]
+
+[id=introduction]
+== Introduction
+
+I had been using a https://www.smappee.com/infinity[_Smappee_] system that came with the solar panel installation at home for about two years. It was installed back in July 2020 to monitor the electricity generation and consumption in our house when I noticed something odd. I used the neat _Smappee_ iOS app to check the five-minute averages to monthly averages of the energy generated by the solar panels on our roof. The system was tracking 5 minute averages for longer than two weeks until it abruptly reduced the latter to two weeks. Data older than two weeks was aggregated at a longer one hour interval.
```
[the remark this answers is not in the document]
Comment (eloots) [reply posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc:12]: Perfect!

**Comment by eloots** [comment]
> This is a BIG post... 

I get your point @thinkmorestupidless !

On the other hand, splitting it now would take quite a bit of work, so I'm going to stick with the current bigger article. I'm going to add some sort of TOC (or more like an overview of the structure of the article).

File: posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc (hunk 9 above)
Comment (eloots) [reply posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc]: Done!
